src/main/java/com/aws/greengrass/componentmanager/builtins/S3Downloader.java

```diff
@@ -94,6 +96,23 @@ protected long download(long rangeStart, long rangeEnd, MessageDigest messageDig
             }, "download-S3-artifact", logger);
         } catch (InterruptedException e) {
             throw e;
+        } catch (S3Exception e) {
+            if (e.statusCode() == HttpStatusCode.FORBIDDEN) {
+                throw new PackageDownloadException(
+                        getErrorString("Failed to download object from S3. GetObject returns 403 Access Denied"
+                                + "Please check that core device's token exchange role grants s3:GetObject"),
+                        e).withErrorContext(e.getClass().getSimpleName(),
+                        DeploymentErrorCode.S3_GET_BUCKET_ACCESS_DENIED);
+            }
+            if (e.statusCode() == HttpStatusCode.NOT_FOUND) {
+                throw new PackageDownloadException(getErrorString(
+                        "Failed to download object from S3. GetObject returns 404 Resource Not Found"
```

Comment:
Similar comments about verbose error messages. I think it would be easier to just say:
"GetObject returned _HTTP ERROR CODE + MSG_. Ensure the IAM role associated with the core device has a policy granting s3:GetObject"

Reply:
will update